internal/orchestrator/data_query.go

```diff
@@ -88,6 +88,22 @@ func (or *orchestrator) GetMessageByIDWithData(ctx context.Context, id string) (
 	return or.fetchMessageData(ctx, msg)
 }
 
+func (or *orchestrator) ResolveSigningKey(ctx context.Context, inputKey *core.VerifierRef) (*core.VerifierRef, error) {
+	if inputKey.Type != or.blockchain().VerifierType() {
```

Comment:
One thing to consider is that gateway namespaces don't require a blockchain plugin, so `or.blockchain()` could possibly be `nil`.

Reply:
Checking for it here now: https://github.com/hyperledger/firefly/pull/1038/files#diff-e097a0bac30f72e5cc6978dc4e6336e1377310f01235622293baf87758df9302R153